e2e/single-cluster/gitrepo_test.go

```diff
@@ -51,13 +53,19 @@ var _ = Describe("Monitoring Git repos via HTTP for change", Label("infra-setup"
 
 		inClusterRepoURL = gh.GetInClusterURL(host, port, repoName)
 
-		tmpdir, _ = os.MkdirTemp("", "fleet-")
-		clonedir = path.Join(tmpdir, repoName)
+		tmpDir, _ = os.MkdirTemp("", "fleet-")
+		clonedir = path.Join(tmpDir, repoName)
+
+		gitrepoName = testenv.RandomFilename("gitjob-test", r)
 	})
 
 	AfterEach(func() {
-		os.RemoveAll(tmpdir)
-		_, _ = k.Delete("gitrepo", "gitrepo-test")
+		report := CurrentSpecReport()
+		if report.Failure.IsZero() {
+			os.RemoveAll(tmpDir)
+			_, _ = k.Delete("gitrepo", gitrepoName)
```

Comment:
shouldn't we try to delete the gitrepo even if there was a failure? We need to make sure there is no resources from previous tests

Reply:
That's a good point. I've used this code to keep the resources in case of failures so that I could inspect them. The adaptation of the test makes sure that they names will most likely not be the same, so there shouldn't be any issue with leaving those around for collecting more information about the issue using the `dump-failed-k3ds.sh` script. That being said, I don't have a strong opinion on this one.

Reply:
perhaps resources deployed by a previous test case might affect other test cases. I'd prefer to have a clean state when starting a new test case

Reply:
I prefer the fast cleanup we had before:

* Clean up is not immediate, we can't trust it without extensive waiting.
* I don't want to wait for a proper clean up and fail the test if clean up fails.  
* Tests run in parallel and should not conflict with each other anyhow.
* I would focus and comment the clean up, if I have to inspect a failure.

Reply:
Alright, I'll undo that.